### ISSUE TITLE
Nuxt 4 Support

### ISSUE THREAD

**Issue by offizium-berndstorath** [nuxt/eslint#487]
### Environment

- Operating System: `Darwin`
- Node Version:     `v20.15.0`
- Nuxt Version:     `3.12.4`
- CLI Version:      `3.12.0`
- Nitro Version:    `-`
- Package Manager:  `pnpm@9.7.0`
- Builder:          `-`
- User Config:      `future`, `compatibilityDate`, `devtools`, `modules`, `colorMode`
- Runtime Modules:  `@nuxtjs/i18n@8.3.3`, `@nuxtjs/tailwindcss@6.12.1`, `@pinia/nuxt@0.5.3`, `@eschricht/nuxt-color-mode@1.1.5`
- Build Modules:    `-`

### Package

None

### Reproduction

Package is `@nuxt/eslint-config` issue form is outdated

Use new Nuxt 4 Folder Structure

https://github.com/offizium-berndstorath/eslint-repro

### Describe the bug

Component `Button` inside `app/components/base/Button.vue` throws error `Component name "Button" should always be multi-word.`.
Using the old folder structure this doesn't throw because it shouldnt as the component would be used as `BaseButton` (which is multi word)

### Additional context

Reopened #479

**Comment by antfu** [comment]
You are not using `@nuxt/eslint` - which means the configs are not project-aware. If you decided to do it that way, then you are responsible to config the path correctly. For example

```ts
import { createConfigForNuxt } from '@nuxt/eslint-config/flat';

export default createConfigForNuxt({
  dirs: {
    components: ['app/components'],
    pages: ['app/pages'],
    // ... and the others
  }
})
```
